pkg/uri/parser_test.go

```diff
@@ -18,61 +18,60 @@ func TestParse(t *testing.T) {
 		Err      error
 		Expected *uri.URI
 	}{
-		{"lakefs://foo@bar/baz", nil, &uri.URI{
+		{"lakefs://foo/bar/baz", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 			Ref:        "bar",
 			Path:       strp("baz"),
 		}},
-		{"lakefs://foo@bar/baz/path", nil, &uri.URI{
+		{"lakefs://foo/bar/baz/path", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 			Ref:        "bar",
 			Path:       strp("baz/path"),
 		}},
-		{"lakefs://foo@bar/baz/path@withappendix.foo", nil, &uri.URI{
+		{"lakefs://foo/bar/baz/path@withappendix.foo", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 			Ref:        "bar",
 			Path:       strp("baz/path@withappendix.foo"),
 		}},
-		{"lakefs://fo/o@bar/baz/path@withappendix.foo", nil, &uri.URI{
+		{"lakefs://fo-o/bar/baz/path@withappendix.foo", nil, &uri.URI{
 			Protocol:   "lakefs",
-			Repository: "fo/o",
+			Repository: "fo-o",
 			Ref:        "bar",
 			Path:       strp("baz/path@withappendix.foo"),
 		}},
 		{"lakefs://foo", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 		}},
-		{"lakefs://foo@bar/", nil, &uri.URI{
+		{"lakefs://foo/bar/", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 			Ref:        "bar",
 			Path:       strp(""),
 		}},
-		{"lakefs://foo@bar", nil, &uri.URI{
+		{"lakefs://foo/bar", nil, &uri.URI{
 			Protocol:   "lakefs",
 			Repository: "foo",
 			Ref:        "bar",
 		}},
-		{"lakefssss://foo@bar/baz", uri.ErrMalformedURI, nil},
-		{"lakefs:/foo@bar/baz", uri.ErrMalformedURI, nil},
-		{"lakefs//foo@bar/baz", uri.ErrMalformedURI, nil},
+		{"lakefssss://foo/bar/baz", uri.ErrMalformedURI, nil},
+		{"lakefs:/foo/bar/baz", uri.ErrMalformedURI, nil},
+		{"lakefs//foo/bar/baz", uri.ErrMalformedURI, nil},
 	}
```

Comment:
I would leave some of the `lakefs://foo@bar/goo` tests as negative ones.

Reply:
good idea - in our case foo@bar will be a repository name